pkg/policy/policy.go

```diff
@@ -106,6 +107,17 @@ func (s *SearchContext) CallDepth() string {
 	return strconv.Itoa(s.Depth * 2)
 }
 
+// WithLogger returns a shallow copy of the received SearchContext with the
+// logging set to write to 'log'.
+func (s *SearchContext) WithLogger(log io.Writer) *SearchContext {
+	result := *s
+	result.Logging = logging.NewLogBackend(log, "", 0)
+	if result.Trace == TRACE_DISABLED {
```

Comment:
Is this logic correct? We want to enable if it's disabled? Why not enable it regardless?

Reply:
This was just so that if calling (test) code already has trace level at the verbose level, we won't lower the verbosity to 'enabled'.